### PR TITLE
feat(angular-base): Adding Angular generator

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**

### CODE REVIEW

File: packages/teleport-plugin-css/src/index.ts [hunk 1]
```diff
@@ -21,17 +21,23 @@ interface VueStyleChunkConfig {
   chunkName: string
```
Comment (alexnm) [inline packages/teleport-plugin-css/src/index.ts]: is it feasible to use this plugin for JSX based frameworks as well? maybe we can have another config options like `templateType`: 'JSX' | 'HTML'. Just an idea, we shouldn't focus on this now

File: packages/teleport-component-generator-angular/src/angular-mapping.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "elements": {
+    "group": {
+      "elementType": "div"
+    },
+    "navLink": {
```
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/teleport-component-generator-angular/src/angular-mapping.json]: Yeah, forgot to revert it, was testing initially 👍

File: packages/teleport-plugin-angular-base-component/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  DEFAULT_ANGULAR_TEMPLATE_CHUNK_NAME,
+  DEFAULT_ANGULAR_TS_CHUNK_NAME,
+  ANGULAR_CORE_DEPENDENCY,
+  DEFAULT_TS_CHUNK_AFTER,
+} from './constants'
+
+interface AngularPluginConfig {
+  angularTemplateChunkName: string
+  angularComponentChunkName: string
+  tsChukAfter: string[]
```
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/teleport-plugin-angular-base-component/src/index.ts]: My type of typo's, first you mess up and then in rest of the places intellisense takes care of them to reproduce 😅 😂

File: packages/teleport-plugin-vue-base-component/src/index.ts [hunk 4]
```diff
@@ -50,6 +50,7 @@ export const createPlugin: ComponentPluginFactory<VueComponentConfig> = (config)
           const iterator = useIndex ? `(${iteratorName}, index)` : iteratorName
           return `${iterator} in ${iteratedCollection}`
         },
+        customElementTagNames: (value) => value,
```
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/teleport-plugin-vue-base-component/src/index.ts]: Ok, but in that case shouldn't we export components in that namespace?

File: packages/teleport-plugin-css/src/index.ts (hunk 1 above)
Comment (JayaKrishnaNamburu) [reply packages/teleport-plugin-css/src/index.ts]: Yeah, I feel we could completely use it. We just need to add a 'flavor` flag and use babel types and traves the tree and inject style tags as we do in react based style flavors 👍  That gives support for CSS by default for all types 🎉

File: packages/teleport-plugin-vue-base-component/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/teleport-plugin-vue-base-component/src/index.ts]: @alexnm

File: packages/teleport-plugin-angular-base-component/src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+} from '@teleporthq/teleport-shared/dist/cjs/utils/uidl-utils'
+import { UIDLPropDefinition, UIDLStateDefinition, ComponentUIDL } from '@teleporthq/teleport-types'
+
+export const generateComponentDecorator = (uidl: ComponentUIDL, t = types) => {
+  const decoratorArgs = [
+    t.objectExpression([
+      t.objectProperty(t.identifier('selector'), t.stringLiteral('app-root')),
+      t.objectProperty(
+        t.identifier('templateUrl'),
+        t.stringLiteral(`./${getComponentFileName(uidl)}.html`)
```
Comment (alexnm) [inline packages/teleport-plugin-angular-base-component/src/utils.ts]: forgot about this, but please use FILE_TYPE.HTML and FILE_TYPE.css for `html` and `css` to have consistency across the entire project

File: packages/teleport-shared/src/node-handlers/node-to-html/index.ts [hunk 6]
```diff
@@ -63,7 +63,10 @@ const generateElementNode = (
 ) => {
   const { dependencies, templateLookup } = params
   const { elementType, name, key, children, attrs, dependency, events } = node.content
-  const htmlNode = createHTMLNode(elementType)
+  const htmlNode =
+    dependency && dependency.type === 'local'
```
Comment (alexnm) [inline packages/teleport-shared/src/node-handlers/node-to-html/index.ts]: I think it's enough check for dependency, dependency type package should also count as custom element

**Pull request merged**